src/modules/BootStorage.rb

```diff
@@ -275,6 +274,25 @@ def select_ancestors(device, &predicate)
       end
       results
     end
+
+    # Handle an "unknown device" error: Raise an appropriate exception.
+    # @param dev_name [String]
+    def handle_unknown_device(dev_name)
+      # rubocop:disable Style/GuardClause
+      #
+      # I flatly refuse to make my code LESS readable because of a third-rate
+      # check tool. This is the CLASSIC use case for if...else, even if this
+      # mindless rubocop thinks otherwise.
+      #
+      # 2019-02-06 shundhammer
+
+      if dev_name =~ %r{/by-path/} # bsc#1122008, bsc#1116305
+        raise ::Bootloader::BrokenByPathDeviceName, dev_name
+      else
+        raise ::Bootloader::BrokenConfiguration, "Unknown device #{dev_name}"
+      end
+      # rubocop:enable Style/GuardClause
+    end
```

Comment:
I guess rubocop expects something like:

```
raise ::Bootloader::BrokenByPathDeviceName, dev_name if dev_name =~ %r{/by-path/}

raise ::Bootloader::BrokenConfiguration, "Unknown device #{dev_name}"
```

Or

```
if dev_name =~ %r{/by-path/} # bsc#1122008, bsc#1116305
  raise ::Bootloader::BrokenByPathDeviceName, dev_name
end

raise ::Bootloader::BrokenConfiguration, "Unknown device #{dev_name}"  
```

Maybe this last version continues being quite readable and rubocops will be fine.